spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractExecutorChannel.java

```diff
@@ -154,7 +156,7 @@ public void run() {
 					triggerAfterMessageHandled(message, ex, interceptorStack);
 				}
 				if (ex instanceof MessagingException) {
-					throw (MessagingException) ex;
+					throw new MessagingExceptionWrapper(message, (MessagingException) ex);
```

Comment:
Well, technically it doesn't have much difference with the previous implementation.
If we consider this only as for an internal use, I'm fine with any name for the exception class.
Although we might mention that in its JavaDocs.